### ISSUE TITLE
Error ingesting GCP resources

### ISSUE THREAD

**Issue by xino12** [cloudquery/cq-provider-gcp#18]
Hello ^^

I'm seeing this error I assume due to the migration to the new sdk:

```
9:14AM DBG panic: runtime error: invalid memory address or nil pointer dereference
9:14AM DBG [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1036cf2]
9:14AM DBG 
9:14AM DBG goroutine 535 [running]:
9:14AM DBG github.com/cloudquery/cq-provider-gcp/resources.resolveComputeBackendServiceCdnPolicyBypassCacheOnRequestHeaders(0x15d6b00, 0xc00058bf00, 0x15be1e0, 0xc00054b1c0, 0xc005119400, 0x148a5d6, 0x2a, 0x9, 0x0, 0x0, ...)
9:14AM DBG 	/home/runner/work/cq-provider-gcp/cq-provider-gcp/resources/compute_backend_services.go:490 +0x52
9:14AM DBG github.com/cloudquery/cq-provider-sdk/provider/schema.ExecutionData.resolveColumns(0xc000408300, 0x15d6d40, 0xc0000bcb00, 0x15ef0c0, 0xc0000d6c80, 0x0, 0x15d6b00, 0xc00058bf00, 0x15be1e0, 0xc00054b1c0, ...)
9:14AM DBG 	/home/runner/go/pkg/mod/github.com/cloudquery/cq-provider-sdk@v0.1.6/provider/schema/execution.go:148 +0x317
9:14AM DBG github.com/cloudquery/cq-provider-sdk/provider/schema.ExecutionData.resolveResourceValues(0xc000408300, 0x15d6d40, 0xc0000bcb00, 0x15ef0c0, 0xc0000d6c80, 0x0, 0x15d6b00, 0xc00058bf00, 0x15be1e0, 0xc00054b1c0, ...)
9:14AM DBG 	/home/runner/go/pkg/mod/github.com/cloudquery/cq-provider-sdk@v0.1.6/provider/schema/execution.go:132 +0xba
9:14AM DBG github.com/cloudquery/cq-provider-sdk/provider/schema.ExecutionData.resolveResources(0xc000408300, 0x15d6d40, 0xc0000bcb00, 0x15ef0c0, 0xc0000d6c80, 0x0, 0x15d6b00, 0xc00058bf00, 0x15be1e0, 0xc00054b1c0, ...)
9:14AM DBG 	/home/runner/go/pkg/mod/github.com/cloudquery/cq-provider-sdk@v0.1.6/provider/schema/execution.go:108 +0x1df
9:14AM DBG github.com/cloudquery/cq-provider-sdk/provider/schema.ExecutionData.callTableResolve(0xc000408300, 0x15d6d40, 0xc0000bcb00, 0x15ef0c0, 0xc0000d6c80, 0x0, 0x15d6b00, 0xc00058bf00, 0x15be1e0, 0xc00054b1c0, ...)
9:14AM DBG 	/home/runner/go/pkg/mod/github.com/cloudquery/cq-provider-sdk@v0.1.6/provider/schema/execution.go:88 +0x232
9:14AM DBG github.com/cloudquery/cq-provider-sdk/provider/schema.ExecutionData.ResolveTable.func1(0x0, 0x0)
9:14AM DBG 	/home/runner/go/pkg/mod/github.com/cloudquery/cq-provider-sdk@v0.1.6/provider/schema/execution.go:48 +0xb3
9:14AM DBG golang.org/x/sync/errgroup.(*Group).Go.func1(0xc0002e2ab0, 0xc000604840)
9:14AM DBG 	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:57 +0x59
9:14AM DBG created by golang.org/x/sync/errgroup.(*Group).Go
9:14AM DBG 	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54 +0x66
```

**Comment by xino12** [comment]
I also see this kind of error when it's fetching gcp_compute_images I think.


```
Error: rpc error: code = Unknown desc = googleapi: got HTTP response code 404 with body: <!DOCTYPE html>
<html lang=en>
  <meta charset=utf-8>
  <meta name=viewport content="initial-scale=1, minimum-scale=1, width=device-width">
  <title>Error 404 (Not Found)!!1</title>
  <style>
    *{margin:0;padding:0}html,code{font:15px/22px arial,sans-serif}html{background:#fff;color:#222;padding:15px}body{margin:7% auto 0;max-width:390px;min-height:180px;padding:30px 0 15px}* > body{background:url(//www.google.com/images/errors/robot.png) 100% 5px no-repeat;padding-right:205px}p{margin:11px 0 22px;overflow:hidden}ins{color:#777;text-decoration:none}a img{border:0}@media screen and (max-width:772px){body{background:none;margin-top:0;max-width:none;padding-right:0}}#logo{background:url(//www.google.com/images/branding/googlelogo/1x/googlelogo_color_150x54dp.png) no-repeat;margin-left:-5px}@media only screen and (min-resolution:192dpi){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat 0% 0%/100% 100%;-moz-border-image:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) 0}}@media only screen and (-webkit-min-device-pixel-ratio:2){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat;-webkit-background-size:100% 100%}}#logo{display:inline-block;height:54px;width:150px}
  </style>
  <a href=//www.google.com/><span id=logo aria-label=Google></span></a>
  <p><b>404.</b> <ins>That’s an error.</ins>
  <p>The requested URL <code>/v1beta1/projects/cloudbees-sdm-staging/location/-/registrations?alt=json&amp;pageToken=&amp;prettyPrint=false</code> was not found on this server.  <ins>That’s all we know.</ins>
```

Maybe a 404 shouldn't stop ingesting?

**Comment by yevgenypats** [comment]
Hi, the first error should be fixed in [v0.2.1](https://github.com/cloudquery/cq-provider-gcp/releases/tag/v0.2.1). Yeah, it's probably  due to the migration. We've added unit-tests but possibly this bug was missed. We will also add e2e tests so it should make things more stable.

Looking into the not found response

**Comment by xino12** [comment]
Thanks for the super fast response @yevgenypats really appreciate it. I'm testing the fix now. I'll let you know if the seg fault does not happen anymore.
